torch_xla/csrc/tensor.cpp

```diff
@@ -792,6 +792,20 @@ void XLATensor::SetSubView(ViewInfo view_info) const {
   data()->generation += 1;
 }
 
+void XLATensor::ModifyCurrentView(ViewInfo view_info) const {
+  if (data()->view != nullptr) {
+    data()->view = data()->view->CreateSubView(view_info.shape, view_info);
+    return;
+  }
+  // This node is not a view. Since this function is meant to modify a view
+  // in place, we need to turn this existing tensor into a view.
+  ir::Value ir_value = GetIrValue();
+  std::shared_ptr<Alias> alias = std::make_shared<Alias>(ir_value);
+  data()->view =
+      std::make_shared<View>(ir_value.shape(), alias, std::move(view_info));
```

Comment:
@bdhirsh can you explain why you used ir_value.shape() here instead of view_info.shape, when initializing the shape of the View op?

Reply:
I think you're right, this looks like it should take in `view_info.shape`. I probably pulled this from [here](https://github.com/pytorch/xla/pull/3018/files#diff-17ef64dc5ef931bed0cb0d971cf537a610e90a9b4351658d6587ff59448f0f87R820) (which is different because it's creating a "noop" view).